pyriemann/transfer/pipelines.py

```diff
@@ -0,0 +1,822 @@
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y_enc : ndarray, shape (n_matrices,)
+            Extended labels for each matrix.
+
+        Returns
+        -------
+        self : TLEstimator instance
+            The TLEstimator instance.
+        """
+
+        X_dec, y_dec, domains = decode_domains(X, y_enc)
+
+        if is_regressor(self.estimator):
+            y_dec = y_dec.astype(float)
+
+        if self.domain_weight is not None:
+            w = np.zeros(len(X_dec))
+            for d in np.unique(domains):
+                w[domains == d] = self.domain_weight[d]
+        else:
+            w = None
+
+        if isinstance(self.estimator, Pipeline):
+            sample_weight = {}
+            for step in self.estimator.steps:
+                step_name = step[0]
+                sample_weight[step_name + '__sample_weight'] = w
+            self.estimator.fit(X_dec, y_dec, **sample_weight)
+        else:
+            self.estimator.fit(X_dec, y_dec, sample_weight=w)
+
+        return self
+
+    def predict(self, X):
+        """Get the predictions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray, shape (n_matrices,)
+            Predictions for each matrix according to the estimator
+        """
+        return self.estimator.predict(X)
+
+    def score(self, X, y_enc):
+        """Return the score of self.estimator
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y_enc : ndarray, shape (n_matrices,)
+            Extended labels for each matrix.
+
+        Returns
+        -------
+        score : float
+            If self.estimator is a classifier, then return the mean accuracy on
+            the given test data and labels. If self.estimator is a regressor,
+            return the coefficient of determination of the prediction.
+        """
+        _, y_true, _ = decode_domains(X, y_enc)
+        y_pred = self.predict(X)
+        if is_classifier(self.estimator):
+            return accuracy_score(y_true, y_pred)
+        elif is_regressor(self.estimator):
+            y_true = y_true.astype(float)
+            return r2_score(y_true, y_pred)
+
+
+class MDWM(MDM):
+    """Classification by Minimum Distance to Weighted Mean.
+
+    Classification by nearest centroid. For each of the given classes, a
+    centroid is estimated, according to the chosen metric, as a weighted mean
+    of SPD matrices from the source domain, combined with the class centroid of
+    the target domain [1]_ [2]_.
+    For classification, a given new matrix is attibuted to the class whose
+    centroid is the nearest according to the chosen metric.
+
+    Parameters
+    ----------
+    transfer_coef : float
```

Comment:
To highlight the link with `domain_weight` of `TLEstimator`,
`transfer_coef` could be renamed `domain_tradeoff`, or `target_domain_weight`.
@sylvchev